pkgs/applications/office/libreoffice/wrapper.nix

```diff
@@ -1,33 +1,101 @@
-{ lib, runCommand
-, libreoffice, dbus, bash, substituteAll
-, coreutils, gnugrep
-, dolphinTemplates ? true
+{ lib
+, stdenv
+# The unwrapped libreoffice derivation
+, unwrapped
+, makeWrapper
+, runCommand
+, substituteAll
+# For Emulating wrapGAppsHook
+, gsettings-desktop-schemas
+, hicolor-icon-theme
+, dconf
+, librsvg
+, gdk-pixbuf
+# Configuration options for the wrapper
+, extraMakeWrapperArgs ? []
+, dbusVerify ? stdenv.isLinux
+, dbus
 }:
-runCommand libreoffice.name {
-  inherit (libreoffice) jdk meta;
-  inherit coreutils dbus gnugrep libreoffice bash;
+
+let
+  makeWrapperArgs = builtins.concatStringsSep " " ([
+    "--set" "GDK_PIXBUF_MODULE_FILE" "${librsvg}/${gdk-pixbuf.moduleDir}.cache"
+    "--prefix" "GIO_EXTRA_MODULES" ":" "${lib.getLib dconf}/lib/gio/modules"
+    "--prefix" "XDG_DATA_DIRS" ":" "${unwrapped.gtk3}/share/gsettings-schemas/${unwrapped.gtk3.name}"
+    "--prefix" "XDG_DATA_DIRS" ":" "$out/share"
+    "--prefix" "XDG_DATA_DIRS" ":" "${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}"
+    "--prefix" "XDG_DATA_DIRS" ":" "${hicolor-icon-theme}/share"
+    "--prefix" "GST_PLUGIN_SYSTEM_PATH_1_0" ":"
+      "${lib.makeSearchPath "lib/girepository-1.0" unwrapped.gst_packages}"
+  ] ++ lib.optionals unwrapped.kdeIntegration [
+    "--prefix" "QT_PLUGIN_PATH" ":" "${
+      lib.makeSearchPath
+      unwrapped.qtbase.qtPluginPrefix
+      (builtins.map lib.getBin unwrapped.qtPackages)
+    }"
+    "--prefix" "QML2_IMPORT_PATH" ":" "${
+      lib.makeSearchPath unwrapped.qtbase.qtQmlPrefix
+      (builtins.map lib.getBin unwrapped.qmlPackages)
+    }"
+  ] ++ [
+    # Add dictionaries from all NIX_PROFILES
+    "--run" (lib.escapeShellArg ''
+      for PROFILE in $NIX_PROFILES; do
+          HDIR="$PROFILE/share/hunspell"
+          if [ -d "$HDIR" ]; then
+              export DICPATH=$DICPATH''${DICPATH:+:}$HDIR
+          fi
+      done
+    '')
+  ] ++ lib.optionals dbusVerify [
+    # If no dbus is running, start a dedicated dbus daemon
+    "--run" (lib.escapeShellArg ''
+      if ! ( test -n "$DBUS_SESSION_BUS_ADDRESS" ); then
```

Comment:
I have the feeling that writing this into a script file will make things more manageable

Reply:
> I have the feeling that writing this into a script file will make things more manageable

I think this is more elegant this way. In any case, I think this should also be the maintainer's decision.